### PR TITLE
Prevent remove from (sometimes) crashing when there's nothing to remove

### PR DESCRIPTION
**Pull request by JohnRusk**

### CODE REVIEW

File: cmd/removeProcessor.go [hunk 1]
```diff
@@ -40,7 +40,11 @@ func newRemoveTransferProcessor(cca *cookedCopyCmdArgs, numOfTransfersPerPart in
 		LogLevel: cca.logVerbosity,
 	}
 
-	reportFirstPart := func() { cca.waitUntilJobCompletion(false) }
+	reportFirstPart := func(jobStarted bool) {
+		if jobStarted {
+			cca.waitUntilJobCompletion(false)
```
Comment (JohnRusk) [inline cmd/removeProcessor.go]: It's a bit ugly that we have three different ways of starting progress reporting.

* "Remove" does it here (and now does it _only_ if there's actually something to remove)
* "Sync" does it all the time (and, unlike Copy and Remove, avoids job lookup errors if there was nothing to remove, by testing a flag on the cooked sync args in its progress reporting routine.  Copy (and therefore remove) have no such flag)
* "Copy" uses code in copyEnumeratorInit which only starts progress reporting if it knows that there is work to do.

I've decided that we probably just have to leave all three of the above in place for now. There's already a TODO on Remove to refactor it to use the copy approach at some time in the future.

Comment (zezha-msft) [reply cmd/removeProcessor.go]: Sounds good.

Comment (adreed-msft) [inline cmd/removeProcessor.go]: Sounds good to me.

Comment (adreed-msft) [reply cmd/removeProcessor.go]: Sounds good to me.

**Pull request merged**